### ISSUE TITLE
Yolo  Bboxes will generate tags larger than 1.

### ISSUE THREAD

**Issue by sang-yc** [albumentations-team/albumentations#862]
Bboxes will generate tags larger than 1 when using 'albumentations' for data augmentation. Why? What should be done?

The error is as follows:
ValueError: Expected x_max for bbox (0.284, 0.15315315315315314, 1.002, 0.993993993993994, 6.0) to be in the range [0.0, 1.0], got 1.002.

**Comment by sang-yc** [comment]
I have found a solution to this problem.
Before data augmentation, the length and width of GT bboxes are subtracted by one pixel, which has little effect on the label bounding box, but it can prevent the overflow of values and prevent the generation of labels greater than 1.


image.shape[0]:  width of image
image.shape[1]:  length of image
bboxes: format yolo 
              [[x, y, w, h, class], [x, y, w, h, class], ...]

`for i in range(len(bboxes)):
            bboxes[i][2] = np.abs(bboxes[i][2] - 0.5 / image.shape[0])
            bboxes[i][3] = np.abs(bboxes[i][3] - 0.5 / image.shape[1])`

To further reduce the impact of processing, I only subtracted 0.5 pixels from the code.

**Comment by sang-yc** [comment]
I have found a solution to this problem.
Before data augmentation, the length and width of GT bboxes are subtracted by one pixel, which has little effect on the label bounding box, but it can prevent the overflow of values and prevent the generation of labels greater than 1.


image.shape[0]:  width of image
image.shape[1]:  length of image
bboxes: format yolo 
              [[x, y, w, h, class], [x, y, w, h, class], ...]
```python
for i in range(len(bboxes)):
            bboxes[i][2] = np.abs(bboxes[i][2] - 0.5 / image.shape[0])
            bboxes[i][3] = np.abs(bboxes[i][3] - 0.5 / image.shape[1])
```
To further reduce the impact of processing, I only subtracted 0.5 pixels from the code.

**Comment by sang-yc** [comment]
I have found a solution to this problem.
Before data augmentation, the length and width of GT bboxes are subtracted by one pixel, which has little effect on the label bounding box, but it can prevent the overflow of values and prevent the generation of labels greater than 1.

labels format: yolo
image.shape[0]:  width of image
image.shape[1]:  length of image
bboxes: [[x, y, w, h, class], [x, y, w, h, class], ...]

```python
        for i in range(len(bboxes)):
            bboxes[i][2] = np.abs(bboxes[i][2] - 0.5 / image.shape[0])
            bboxes[i][3] = np.abs(bboxes[i][3] - 0.5 / image.shape[1])
```

To further reduce the impact of processing, I only subtracted 0.5 pixels from the code.

**Comment by sang-yc** [comment]
I have found a solution to this problem.
Before data augmentation, the length and width of GT bboxes are subtracted by one pixel, which has little effect on the label bounding box, but it can prevent the overflow of values and prevent the generation of labels greater than 1.

labels format: yolo
image.shape[0]: width of image
image.shape[1]: length of image
bboxes: [[x, y, w, h, class], [x, y, w, h, class], ...]

```python
for i in range(len(bboxes)):
    bboxes[i][2] = np.abs(bboxes[i][2] - 0.5 / image.shape[0])
    bboxes[i][3] = np.abs(bboxes[i][3] - 0.5 / image.shape[1])
```
To further reduce the impact of processing, I only subtracted 0.5 pixels from the code.

**Comment by christian-cahig** [comment]
> Bboxes will generate tags larger than 1 when using 'albumentations' for data augmentation. Why? What should be done?
> 
> The error is as follows:
> ValueError: Expected x_max for bbox (0.284, 0.15315315315315314, 1.002, 0.993993993993994, 6.0) to be in the range [0.0, 1.0], got 1.002.

I raised an issue (https://github.com/albumentations-team/albumentations/issues/883) that could possibly explain this.

**Comment by Dipet** [comment]
Should be fixed by #924
